src/ServiceFabric/ServiceFabric.Test/ScenarioTests/ServiceFabricManagedClustersTests.ps1

```diff
@@ -143,4 +143,90 @@ function Test-CertAndExtension
 
 	$removeResponse = $cluster | Remove-AzServiceFabricManagedCluster -PassThru
 	Assert-True { $removeResponse }
+}
+
+# new network security rule test
+function Test-AddNetworkSecurityRule
+{
+	$resourceGroupName = "sfmcps-rg-" + (getAssetname)
+	$clusterName = "sfmcps-" + (getAssetname)
+	$pass = (ConvertTo-SecureString -AsPlainText -Force "TestPass1234!@#")
+	$location = "southcentralus"
+	$testClientTp = "123BDACDCDFB2C7B250192C6078E47D1E1DB119B"
+	Assert-ThrowsContains { Get-AzServiceFabricManagedCluster -ResourceGroupName $resourceGroupName -Name $clusterName } "NotFound"
+
+	$tags = @{"test"="tag"}
+
+	$cluster = New-AzServiceFabricManagedCluster -ResourceGroupName $resourceGroupName -ClusterName $clusterName -Location $location `
+		-AdminPassword $pass -Sku Basic -ClientCertThumbprint $testClientTp -Tag $tags -Verbose
+	Assert-AreEqual "Succeeded" $cluster.ProvisioningState 
+	Assert-AreEqual "Automatic" $cluster.ClusterUpgradeMode
+
+	$pnt = New-AzServiceFabricManagedNodeType -ResourceGroupName $resourceGroupName -ClusterName $clusterName -Name pnt -InstanceCount 5 -DiskType Standard_LRS -Primary
+	Assert-AreEqual 5 $pnt.VmInstanceCount
+	Assert-AreEqual "Standard_LRS" $pnt.DataDiskType
+
+
+	$clusterFromGet = Get-AzServiceFabricManagedCluster -ResourceGroupName $resourceGroupName -Name $clusterName
+	Assert-AreEqual "Ready" $clusterFromGet.ClusterState
+
+	$NSRName = "testSecRule1"
+	$access = "allow"
+	$direction = "inbound"
+	$protocol = "tcp"
+	$sourcePortRanges = "1-1000"
+	$destinationPortRanges = "1-65535"
+	$destinationAddressPrefixes = "194.69.104.0/25", "194.69.119.64/26", "167.220.249.128/26", "255.255.255.255/32"
+	$sourceAddressPrefixes = "167.220.242.0/27", "167.220.0.0/23", "131.107.132.16/28", "167.220.81.128/26"
+
+	$cluster = Add-AzServiceFabricManagedNetworkSecurityRule -ResourceGroupName $resourceGroupName -ClusterName $clusterName `
+			-Name $NSRName -Access $access -Direction $direction -Protocol $protocol -Priority 1200 -SourcePortRanges $sourcePortRanges -DestinationPortRanges $destinationPortRanges -DestinationAddressPrefixes $destinationAddressPrefixes -SourceAddressPrefixes $sourceAddressPrefixes -Verbose
+
+	$clusterFromGet = Get-AzServiceFabricManagedCluster -ResourceGroupName $resourceGroupName -Name $clusterName
+
+	Assert-NotNull $clusterFromGet.NetworkSecurityRules
```

Comment:
probably should check more things here as technically any parameter could get messed up as the user is passing them all individually

Reply:
updated